doc/gambit.txi

```diff
@@ -245,6 +246,122 @@ particular, the macOS and Microsoft Windows prebuilt installers set
 up the environment so that the @code{gcc} compiler finds these files
 automatically.
 
+@node Extending the core system, GSI, Accessing the system files, The Gambit system
+@section Extending the core system
+
+Identifiers with a @code{##} prefix are not valid identifiers according
+to RnRS. This means that code containing @code{##} prefixed identifiers
+cannot be processed by and shared with other Scheme implementations.
+They are hard to read by people that aren't used to that extension.
+Moreover the code lacks abstraction and safety because using
+@code{##car} rather than @code{car} has a specific meaning: avoiding
+type checks. Consequently it is hard to "turn on" safe execution of the
+code when it needs to be debugged. Many parts of the runtime library are
+expressed at a low level of abstraction (with @code{##} prefixed
+identifiers) even when not required.
+
+For those reasons @code{##} prefixed identifiers should be used
+sparingly in new code, and existing code should gradually be rewritten
+to avoid them. The primitive operations which are used to build
+higher-level operations are all defined as procedures with a @code{##}
+prefix. The purpose of the @code{primitive} special form is to abstract
+from the @code{##} prefix. The call @code{(primitive foo)} is equivalent
+to @code{##foo} and @code{(primitive (foo a b c))} is equivalent to
+@code{(##foo a b c)}. Another Scheme implementation could define
+@code{primitive} differently to use a different prefix (or no prefix) or
+implement an arbitrary mapping to implementation dependent identifiers.
+
+The @code{primitive} special form works in tandem with the
+@code{define-procedure} and @code{define-primitive} forms and the
+@code{~~lib/gambit/prim/prim#.scm} file and @code{gambit/prim} library.
```

Comment:
Probably best to use the library name `(gambit prim)` to conform to R7RS.  Also a few lines below.